src/insert-extension.js

```diff
@@ -34,14 +34,25 @@ export function resetExtensionScriptInsertedOrPresentForTesting() {
 /**
  * Check script info in HTML head and make update if necessary
  * @param {!Window} win
- * @param {!Element} element
  * @param {string} extension
+ * @param {boolean} opt_ignoreElementExistenceCheck if true, will not check and
+ *    require that extension element exist in document
  */
-export function insertAmpExtensionScript(win, element, extension) {
+export function insertAmpExtensionScript(
+    win, extension, opt_ignoreElementExistenceCheck) {
   if (extension == 'amp-embed') {
     extension = 'amp-ad';
   }
-  if (isAmpExtensionScriptRequired(win, element, extension)) {
+  if (!opt_ignoreElementExistenceCheck) {
```

Comment:
I mean this entire querying isn't saving us. We know we're have a `<amp-extension>` on the page, because something (the `<amp-extension>`, or `#addCreativeExtensions`) called `insertAmpExtensionScript`. It's like a tautology.